doc/source/cluster/config.rst

```diff
@@ -0,0 +1,139 @@
+
+    # The maximum number of workers nodes to launch in addition to the head
+    # node.
+    max_workers: 0 # this means zero workers
+
+    # Cloud-provider specific configuration.
+    provider:
+       type: aws
+       region: us-west-2
+       availability_zone: us-west-2a
+
+    # How Ray will authenticate with newly launched nodes.
+    auth:
+       ssh_user: ubuntu
+
+    setup_commands:
+      - pip install ray[all]
+      # The following line demonstrate that you can specify arbitrary
+      # startup scripts on the cluster.
+      - touch /tmp/some_file.txt
+
+Most of the example YAML file is optional. Here is a `reference minimal YAML file <https://github.com/ray-project/ray/tree/master/python/ray/autoscaler/aws/example-minimal.yaml>`__, and you can find the defaults for `optional fields in this YAML file <https://github.com/ray-project/ray/tree/master/python/ray/autoscaler/aws/example-full.yaml>`__.
+
+In another example, the `AWS example configuration file <https://github.com/ray-project/ray/tree/master/python/ray/autoscaler/aws/example-full.yaml>`__ cluster config file will create a small cluster with an m5.large head node (on-demand) configured to autoscale up to two m5.large `spot workers <https://aws.amazon.com/ec2/spot/>`__.
+
+You are encouraged to copy the example YAML file and modify it to your needs. This may include adding additional setup commands to install libraries or sync local data files.
+
+Setup Commands
+--------------
+
+.. note:: After you have customized the nodes, it is also a good idea to create a new machine image (or docker container) and use that in the config file. This reduces worker setup time, improving the efficiency of auto-scaling.
+
+The setup commands you use should ideally be *idempotent*, that is, can be run more than once. This allows Ray to update nodes after they have been created. You can usually make commands idempotent with small modifications, e.g. ``git clone foo`` can be rewritten as ``test -e foo || git clone foo`` which checks if the repo is already cloned first.
+
+Docker Support
+--------------
+
+The ``worker_image`` and ``pull_before_run`` fields override the correpsonding field in the top level ``docker`` section for the node type. The ``worker_run_options`` field is combined with top level ``docker: run_options`` field to produce the docker run command for the given node_type. The following configuration is for a GPU enabled node type. Ray will automatically select the Nvidia docker runtime if it is available.
+
+.. code-block:: yaml
+
+    pull_before_run: True
+    worker_image:
+        - rayproject/ray-ml:latest-gpu
+
+Common cluster configurations
+-----------------------------
+
+The ``example-full.yaml`` configuration is enough to get started with Ray, but for more compute intensive workloads you will want to change the instance types to e.g. use GPU or larger compute instance by editing the yaml file. Here are a few common configurations:
```

Comment:
Would it make sense to add a note here mentioning that we use `AWS` in the examples but these solutions are generic.

Reply:
good point, fixed